packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -548,67 +558,83 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
+                                      : defaultUIMetadata)
+                                      .map((p) => (
+                                        <MenuItem
+                                          className='gmw-menu-item'
+                                          data-testid={`gmw-edit-${p.name}`}
+                                          key={p.name}
+                                          onClick={async () => onModify(g, p.name)}
+                                          icon={p.icon}
+                                        >
+                                          {p.displayLabel}
+                                        </MenuItem>
+                                      ))
+                                  }
+                                >
+                                  Edit
+                                </MenuItem>,
+                                <MenuItem
+                                  key={1}
+                                  onClick={async () => openProperties(g)}
+                                  icon={<SvgList />}
+                                  data-testid="gmw-context-menu-item"
+                                >
+                                  Properties
+                                </MenuItem>,
+                                <MenuItem
+                                  key={2}
+                                  onClick={() => {
+                                    setSelectedGroup(g);
+                                    setShowDeleteModal(true);
+                                    close();
                                   }}
-                                />
-                                Edit
-                              </MenuItem>,
-                              <MenuItem
-                                key={1}
-                                onClick={async () => openProperties(g)}
-                                icon={<SvgList />}
-                              >
-                                Properties
-                              </MenuItem>,
-                              <MenuItem
-                                key={2}
-                                onClick={() => {
-                                  setSelectedGroup(g);
-                                  setShowDeleteModal(true);
-                                  close();
-                                }}
-                                icon={<SvgDelete />}
-                              >
-                                Remove
-                              </MenuItem>,
-                            ]}
+                                  icon={<SvgDelete />}
+                                  data-testid="gmw-context-menu-item"
+                                >
+                                  Remove
+                                </MenuItem>,
+                              ].concat(
+                                contextUIs.map((p) => {
+                                  return <MenuItem
```

Comment:
Reduce `<MenuItem />` duplication. Create an array with all the required values and use `map` to create all the `MenuItem` elements.

Reply:
The onClick callback for each MenuItem is quite different and the first one has sub menu items. I think it is more clear to keep it like this now